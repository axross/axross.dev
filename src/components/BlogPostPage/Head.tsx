import NextHead from "next/head";
import * as React from "react";
import { MY_JOB_TITLE, MY_NAME, MY_SOCIAL_MEDIA_LINKS } from "../../constant/data";
import { PROFILE_IMAGE_PATH } from "../../constant/staticFilePaths";
import BlogPost from "../../entities/BlogPost";
import useLocale from "../../hooks/useLocale";
import useCanonicalURL from "./useCanonicalURL";
import useTranslation from "../../hooks/useTranslation";

interface Props {
  blogPost?: BlogPost | null;
  blogPostLoading?: boolean;
}

export default function Head({ blogPost, blogPostLoading }: Props) {
  const { availableLocales, currentLocale } = useLocale();
  const canonicalURL = useCanonicalURL();
  const websiteTitle = useTranslation("WEBSITE_TITLE");
  const titleLoading = useTranslation("WEBSITE_TITLE_BLOG_POST_LOADING");
  const titleNotFound = useTranslation("WEBSITE_TITLE_BLOG_POST_NOT_FOUND");
  const title = useTranslation("WEBSITE_TITLE_BLOG_POST", { title: blogPost?.title });
  const profileImageURL = new URL(PROFILE_IMAGE_PATH, canonicalURL.origin);

  if (blogPostLoading) {
    return (
      <NextHead>
        <title>
          {titleLoading}
        </title>
      </NextHead>
    );
  }

  if (!blogPost) {
    return (
      <NextHead>
        <title>
          {titleNotFound}
        </title>
      </NextHead>
    );
  }

  return (
    <NextHead>
      <title>{title}</title>
      <meta name="description" content={blogPost.summary} key="description" />
      <link rel="canonical" href={canonicalURL.href} key="canonical" />

      {availableLocales
        .filter(locale => locale !== currentLocale)
        .map(locale => {
          const url = new URL(canonicalURL.href);
          url.searchParams.set("hl", currentLocale);

          return ( 
            <link
              rel="alternate"
              hrefLang={locale}
              href={url.href}
              key={`alternate:${locale}`}
            />
          );
        })
      }

      {(() => {
        const feedURL = new URL("/posts/feed.xml", canonicalURL.origin);
        feedURL.searchParams.set("hl", currentLocale);

        return (
          <link
            rel="alternate"
            type="application/atom+xml"
            title={`Blog post Atom feed (${currentLocale})`}
            href={feedURL.href}
            key={`atomFeed:${currentLocale}`}
          />
        );
      })()}

      <script type="application/ld+json" key="linked-data">
        {JSON.stringify({
          "@context": "https://schema.org",
          "@type": "BlogBlogPosting",
          url: canonicalURL.href,
          name: blogPost.title,
          headline: blogPost.title,
          description: blogPost.summary,
          thumbnailUrl: profileImageURL,
          image: profileImageURL,
          myself: {
            "@type": "Person",
            name: MY_NAME,
            image: profileImageURL,
            jobTitle: MY_JOB_TITLE,
            sameAs: MY_SOCIAL_MEDIA_LINKS.map(({ url }) => `${url}`)
          },
          copyrightHolder: {
            "@type": "Person",
            name: MY_NAME,
            image: profileImageURL,
            jobTitle: MY_JOB_TITLE,
            sameAs: MY_SOCIAL_MEDIA_LINKS.map(({ url }) => `${url}`)
          },
          copyrightYear: `${new Date().getFullYear}`,
          dateCreated: blogPost.createdAt.toISOString(),
          datePublished: blogPost.createdAt.toISOString(),
          dateModified: blogPost.lastModifiedAt.toISOString(),
          mainEntityOfPage: canonicalURL.href
        })}
      </script>

      <meta property="og:url" content={canonicalURL.href} key="og:url" />
      <meta property="og:type" content="article" key="og:type" />
      <meta
        property="og:description"
        content={blogPost.summary}
        key="og:description"
      />
      <meta property="og:locale" content={currentLocale} key="og:locale" />
      {availableLocales
        .filter(locale => locale !== currentLocale)
        .map(locale => (
          <meta
            property="og:locale:alternate"
            content={locale}
            key={`og:locale:${locale}`}
          />
        ))}
      <meta
        property="og:site_name"
        content={websiteTitle}
        key="og:site_name"
      />
      <meta property="og:title" content={title} key="og:title" />
      <meta
        property="og:image"
        content={profileImageURL.href}
        key="og:image"
      />
    </NextHead>
  );
}