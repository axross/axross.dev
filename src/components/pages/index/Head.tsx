import NextHead from "next/head";
import * as React from "react";
import IntlMessageFormat from "intl-messageformat";
import { MY_JOB_TITLE, MY_NAME, MY_SOCIAL_MEDIA_LINKS } from "../../../constant/data";
import { PROFILE_IMAGE_PATH } from "../../../constant/staticFilePaths";
import { WEBSITE_DESCRIPTION, WEBSITE_TITLE } from "../../../dictionary";
import useLocale from "../../../hooks/useLocale";
import useCanonicalURL from "./useCanonicalURL";

export default function Head() {
  const { availableLocales, currentLocale } = useLocale();
  const canonicalURL = useCanonicalURL();
  const profileImageURL = new URL(PROFILE_IMAGE_PATH, process.env.ORIGIN);

  const title = new IntlMessageFormat(WEBSITE_TITLE[currentLocale]).format();
  const description = new IntlMessageFormat(WEBSITE_DESCRIPTION[currentLocale]).format();
  
  return (
    <NextHead>
      <title>{title}</title>
      <meta name="description" content={description} key="description" />
      <link rel="canonical" href={canonicalURL.href} key="canonical" />

      {availableLocales
        .filter(locale => locale !== currentLocale)
        .map(locale => {
          const url = new URL(canonicalURL.href);
          url.searchParams.set("hl", locale);

          return ( 
            <link
              rel="alternate"
              hrefLang={locale}
              href={url.href}
              key={`alternate:${locale}`}
            />
          );
        })
      }

      {(() => {
        const url = new URL("/posts/feed.xml", process.env.ORIGIN);
        url.searchParams.set("hl", currentLocale);

        return (
          <link
            rel="alternate"
            type="application/atom+xml"
            title={`Blog post Atom feed (${currentLocale})`}
            href={url.href}
            key={`atomFeed:${currentLocale}`}
          />
        );
      })()}

      <script type="application/ld+json" key="linked-data">
        {JSON.stringify({
          "@context": "https://schema.org",
          "@type": "Person",
          url: canonicalURL.href,
          name: MY_NAME,
          image: profileImageURL.href,
          jobTitle: MY_JOB_TITLE,
          sameAs: MY_SOCIAL_MEDIA_LINKS.map(({ url }) => `${url}`)
        })}
      </script>

      <meta property="og:url" content={canonicalURL.href} key="og:url" />
      <meta property="og:type" content="profile" key="og:type" />
      <meta
        property="og:description"
        content={description}
        key="og:description"
      />
      <meta property="og:locale" content={currentLocale} key="og:locale" />
      {availableLocales
        .filter(locale => locale !== currentLocale)
        .map(locale => (
          <meta
            property="og:locale:alternate"
            content={locale}
            key={`og:locale:${locale}`}
          />
        ))}
      <meta property="og:site_name" content={title} key="og:site_name" />
      <meta property="og:title" content={title} key="og:title" />
      <meta
        property="og:image"
        content={profileImageURL.href}
        key="og:image"
      />
    </NextHead>
  )
}