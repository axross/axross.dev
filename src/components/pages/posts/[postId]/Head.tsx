import NextHead from "next/head";
import * as React from "react";
import IntlMessageFormat from "intl-messageformat";
import { MY_JOB_TITLE, MY_NAME, MY_SOCIAL_MEDIA_LINKS } from "../../../../constant/data";
import { PROFILE_IMAGE_PATH } from "../../../../constant/staticFilePaths";
import LocaleContext from "../../../../contexts/LocaleContext";
import { WEBSITE_TITLE, WEBSITE_TITLE_BLOG_POST_NOT_FOUND } from "../../../../dictionary";
import BlogPost from "../../../../entities/BlogPost";
import useCanonicalURL from "./useCanonicalURL";

interface Props {
  blogPost: BlogPost | null;
  blogPostLoading?: boolean;
}

export default function Head({ blogPost, blogPostLoading }: Props) {
  const { availableLocales, currentLocale } = React.useContext(LocaleContext);
  const canonicalURL = useCanonicalURL();
  const profileImageURL = new URL(PROFILE_IMAGE_PATH, process.env.ORIGIN);

  if (blogPostLoading) {
    return (
      <NextHead>
        <title>Loading a blog post...</title>
      </NextHead>
    );
  }

  if (!blogPost) {
    return (
      <NextHead>
        <title>
          {new IntlMessageFormat(
            WEBSITE_TITLE_BLOG_POST_NOT_FOUND[currentLocale]
          ).format()}
        </title>
      </NextHead>
    );
  }

  const title = new IntlMessageFormat(WEBSITE_TITLE[currentLocale]).format({
    name: MY_NAME
  });

  return (
    <NextHead>
      <title>{title}</title>
      <meta name="description" content={blogPost.summary} key="description" />
      <link rel="canonical" href={canonicalURL.href} key="canonical" />

      {availableLocales
        .filter(locale => locale !== currentLocale)
        .map(locale => {
          const url = new URL(canonicalURL.href);
          url.searchParams.set("hl", currentLocale);

          return ( 
            <link
              rel="alternate"
              hrefLang={locale}
              href={url.href}
              key={`alternate:${locale}`}
            />
          );
        })
      }

      {(() => {
        const url = new URL("/posts/feed.xml", process.env.ORIGIN);
        url.searchParams.set("hl", currentLocale);

        return (
          <link
            rel="alternate"
            type="application/atom+xml"
            title={`Blog post Atom feed (${currentLocale})`}
            href={url.href}
            key={`atomFeed:${currentLocale}`}
          />
        );
      })()}

      <script type="application/ld+json" key="linked-data">
        {JSON.stringify({
          "@context": "https://schema.org",
          "@type": "BlogBlogPosting",
          url: canonicalURL.href,
          name: blogPost.title,
          headline: blogPost.title,
          description: blogPost.summary,
          thumbnailUrl: profileImageURL,
          image: profileImageURL,
          myself: {
            "@type": "Person",
            name: MY_NAME,
            image: profileImageURL,
            jobTitle: MY_JOB_TITLE,
            sameAs: MY_SOCIAL_MEDIA_LINKS.map(({ url }) => `${url}`)
          },
          copyrightHolder: {
            "@type": "Person",
            name: MY_NAME,
            image: profileImageURL,
            jobTitle: MY_JOB_TITLE,
            sameAs: MY_SOCIAL_MEDIA_LINKS.map(({ url }) => `${url}`)
          },
          copyrightYear: `${new Date().getFullYear}`,
          dateCreated: blogPost.createdAt.toISOString(),
          datePublished: blogPost.createdAt.toISOString(),
          dateModified: blogPost.lastModifiedAt.toISOString(),
          mainEntityOfPage: canonicalURL.href
        })}
      </script>

      <meta property="og:url" content={canonicalURL.href} key="og:url" />
      <meta property="og:type" content="article" key="og:type" />
      <meta
        property="og:description"
        content={blogPost.summary}
        key="og:description"
      />
      <meta property="og:locale" content={currentLocale} key="og:locale" />
      {availableLocales
        .filter(locale => locale !== currentLocale)
        .map(locale => (
          <meta
            property="og:locale:alternate"
            content={locale}
            key={`og:locale:${locale}`}
          />
        ))}
      <meta
        property="og:site_name"
        content={new IntlMessageFormat(WEBSITE_TITLE[currentLocale]).format({
          name: MY_NAME
        })}
        key="og:site_name"
      />
      <meta property="og:title" content={title} key="og:title" />
      <meta
        property="og:image"
        content={profileImageURL.href}
        key="og:image"
      />
    </NextHead>
  );
}