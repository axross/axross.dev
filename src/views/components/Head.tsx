import IntlMessageFormat from "intl-messageformat";
import NextHead from "next/head";
import * as React from "react";
import { AVAILABLE_LOCALES, DEFAULT_LOCALE } from "../../constant/locale";
import useAvailableLocales from "../hooks/useAvailableLocales";
import useCurrentLocale from "../hooks/useCurrentLocale";
import useMyself from "../hooks/useMyself";
import useSelfUrl from "../hooks/useSelfUrl";
import useTranslation from "../hooks/useTranslation";

interface Props {
  type: string;
  canonicalUrl: URL;
  title: string;
  description: string;
  imageUrl?: string;
  linkingData: Record<string, any>;
}

export default function Head({
  type,
  canonicalUrl,
  title,
  description,
  imageUrl,
  linkingData
}: Props) {
  const url = useSelfUrl();
  const translation = useTranslation();
  const availableLocales = useAvailableLocales();
  const currentLocale = useCurrentLocale();
  const myself = useMyself();
  const alternativeLocales = availableLocales.filter(
    locale => locale !== currentLocale
  );

  return (
    <NextHead>
      <title>{title}</title>
      <meta name="description" content={description} key="description" />
      <link rel="canonical" href={canonicalUrl.href} key="canonical" />
      {alternativeLocales.map(locale => (
        <link
          rel="alternate"
          hrefLang={locale}
          href={buildUrlWithLocale(canonicalUrl, locale).href}
          key={`alternate:${locale}`}
        />
      ))}

      {/* feed */}
      {AVAILABLE_LOCALES.map(locale => (
        <link
          rel="alternate"
          type="application/atom+xml"
          title={`Blog post Atom feed (${locale})`}
          href={
            url.origin +
            "/posts/feed.xml" +
            (locale === DEFAULT_LOCALE ? "" : `?hl=${locale}`)
          }
          key={`atomFeed:${locale}`}
        />
      ))}

      {/* open graph */}
      <meta property="og:url" content={canonicalUrl.href} key="og:url" />
      <meta property="og:type" content={type} key="og:type" />
      <meta
        property="og:description"
        content={description}
        key="og:description"
      />
      <meta property="og:locale" content={currentLocale} key="og:locale" />
      {alternativeLocales.map(locale => (
        <meta
          property="og:locale:alternate"
          content={locale}
          key={`og:locale:${locale}`}
        />
      ))}
      <meta
        property="og:site_name"
        content={new IntlMessageFormat(translation["website.title"]).format({
          screenName: myself.screenName,
          name: myself.name
        })}
        key="og:site_name"
      />
      <meta property="og:title" content={title} key="og:title" />
      <meta
        property="og:image"
        content={imageUrl || url.origin + "/static/profile.jpg"}
        key="og:image"
      />

      {/* json linking data */}
      <script
        type="application/ld+json"
        dangerouslySetInnerHTML={{
          __html: JSON.stringify(linkingData)
        }}
      />
    </NextHead>
  );
}

function buildUrlWithLocale(url: URL, locale: string) {
  const _url = new URL(url.href);

  if (locale === "en-US") {
    _url.searchParams.delete("hl");
  } else {
    _url.searchParams.set("hl", locale);
  }

  return _url;
}
