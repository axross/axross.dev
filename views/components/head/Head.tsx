import NextHead from "next/head";
import * as React from "react";
import Person from "../../../entities/Person";
import useAvailableLocales from "../../hooks/useAvailableLocales";
import useCurrentLocale from "../../hooks/useCurrentLocale";
import useSelfUrl from "../../hooks/useSelfUrl";
import useTranslation from "../../hooks/useTranslation";
import GlobalStyle from "./GlobalStyle";
import GoogleAnalytics from "./GoogleAnalytics";

interface Props {
  person: Person;
  type: string;
  canonicalUrl: URL;
  title: string;
  description: string;
  imageUrl?: string;
  linkingData: Record<string, any>;
}

function Head({
  person,
  type,
  canonicalUrl,
  title,
  description,
  imageUrl,
  linkingData
}: Props) {
  const url = useSelfUrl();
  const translation = useTranslation();
  const availableLocales = useAvailableLocales();
  const currentLocale = useCurrentLocale();
  const alternativeLocales = availableLocales.filter(
    locale => locale !== currentLocale
  );

  return (
    <>
      <NextHead>
        <meta
          name="viewport"
          content="width=device-width,height=device-height"
        />

        <title>{title}</title>
        <meta name="description" content={description} key="description" />
        <link rel="canonical" href={canonicalUrl.href} key="canonical" />
        {alternativeLocales.map(locale => (
          <link
            rel="alternate"
            hrefLang={locale}
            href={buildUrlWithLocale(canonicalUrl, locale).href}
            key={`alternate:${locale}`}
          />
        ))}

        {/* open graph */}
        <meta property="og:url" content={canonicalUrl.href} key="og:url" />
        <meta property="og:type" content={type} key="og:type" />
        <meta
          property="og:description"
          content={description}
          key="og:description"
        />
        <meta property="og:locale" content={currentLocale} key="og:locale" />
        {alternativeLocales.map(locale => (
          <meta
            property="og:locale:alternate"
            content={locale}
            key={`og:locale:${locale}`}
          />
        ))}
        <meta
          property="og:site_name"
          content={translation["website.title"](person)}
          key="og:site_name"
        />
        <meta property="og:title" content={title} key="og:title" />
        <meta
          property="og:image"
          content={imageUrl || url.origin + "/static/profile.jpg"}
          key="og:image"
        />

        {/* json linking data */}
        <script
          type="application/ld+json"
          dangerouslySetInnerHTML={{
            __html: JSON.stringify(linkingData)
          }}
        />

        <GoogleAnalytics />
      </NextHead>

      <GlobalStyle />
    </>
  );
}

function buildUrlWithLocale(url: URL, locale: string) {
  const _url = new URL(url.href);

  if (locale === "en-US") {
    _url.searchParams.delete("hl");
  } else {
    _url.searchParams.set("hl", locale);
  }

  return _url;
}

export default Head;
