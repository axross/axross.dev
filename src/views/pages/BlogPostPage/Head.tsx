import IntlMessageFormat from "intl-messageformat";
import * as React from "react";
import BlogPost from "../../../entities/BlogPost";
import useMyself from "../../hooks/useMyself";
import useSelfUrl from "../../hooks/useSelfUrl";
import useTranslation from "../../hooks/useTranslation";
import Head from "../../components/Head";

interface Props {
  blogPost: BlogPost;
}

export default function BlogPostHead({ blogPost }: Props) {
  const url = useSelfUrl();
  const translation = useTranslation();
  const canonicalUrl = new URL(url.href);
  const myself = useMyself();

  canonicalUrl.searchParams.forEach((value, key) => {
    if (key !== "hl" || value !== "ja-JP") {
      canonicalUrl.searchParams.delete(key);
    }
  });

  const contentHolderLinkingData = {
    "@type": "Person",
    name: myself.name,
    image: url.origin + "/static/profile.jpg",
    jobTitle: myself.jobTitle,
    sameAs: myself.socialLinks.map(({ url }) => url)
  };

  const linkingData = {
    "@context": "https://schema.org",
    "@type": "BlogBlogPosting",
    url: canonicalUrl,
    name: blogPost.title,
    headline: blogPost.title,
    description: blogPost.summary,
    thumbnailUrl: url.origin + "/static/profile.jpg",
    image: url.origin + "/static/profile.jpg",
    myself: contentHolderLinkingData,
    copyrightHolder: contentHolderLinkingData,
    copyrightYear: "2019",
    dateCreated: blogPost.createdAt.toISOString(),
    datePublished: blogPost.createdAt.toISOString(),
    dateModified: blogPost.lastModifiedAt.toISOString(),
    mainEntityOfPage: canonicalUrl.href
  };

  return (
    <Head
      type="article"
      canonicalUrl={canonicalUrl}
      title={new IntlMessageFormat(
        translation["website.title_blog_post"]
      ).format({
        title: blogPost.title,
        name: myself.name,
        screenName: myself.screenName
      })}
      description={blogPost.summary}
      linkingData={linkingData}
    />
  );
}
