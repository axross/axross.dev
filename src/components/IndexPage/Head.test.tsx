import * as React from "react";
import { act, create } from "react-test-renderer";
import MockApp from "../../../fixtures/MockApp";
import Head from "./Head";

describe("<Head>", () => {
  describe("when currentLocale=\"en-US\"", () => {
    it("pushes tags into <head> and they matches with the previous snapshots", async () => {
      const onHeadChange = jest.fn().mockName("onHeadChange");
  
      await act(async () => {
        create(
          <MockApp onHeadChange={onHeadChange} currentLocale="en-US">
            <Head />
          </MockApp>
        );
      });
  
      expect(onHeadChange.mock.calls[0][0]).toMatchSnapshot();
    });
  });

  describe("when currentLocale=\"ja-JP\"", () => {
    it("pushes tags into <head> and they matches with the previous snapshots", async () => {
      const onHeadChange = jest.fn().mockName("onHeadChange");
  
      await act(async () => {
        create(
          <MockApp onHeadChange={onHeadChange} currentLocale="ja-JP">
            <Head />
          </MockApp>
        );
      });
  
      expect(onHeadChange.mock.calls[0][0]).toMatchSnapshot();
    });
  });
});
