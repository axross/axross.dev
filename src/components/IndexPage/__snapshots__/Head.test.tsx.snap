// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Head> when currentLocale="en-US" pushes tags into <head> and they matches with the previous snapshots 1`] = `
Array [
  <meta
    content="width=device-width"
    name="viewport"
  />,
  <meta
    charSet="utf-8"
  />,
  <title>
    Kohei Asai
  </title>,
  <meta
    content="Kohei Asai's personal website. My experience, projects and bloggging."
    name="description"
  />,
  <link
    href="https://tests.kohei.dev/?hl=en-US"
    rel="canonical"
  />,
  <link
    href="https://tests.kohei.dev/?hl=ja-JP"
    hrefLang="ja-JP"
    rel="alternate"
  />,
  <link
    href="https://tests.kohei.dev/posts/feed.xml?hl=en-US"
    rel="alternate"
    title="Blog post Atom feed (en-US)"
    type="application/atom+xml"
  />,
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"Person","url":"https://tests.kohei.dev/?hl=en-US","name":"Kohei Asai","image":"https://tests.kohei.dev/profile.jpg","jobTitle":"Frontend Engineer","sameAs":["https://github.com/axross","https://linkedin.com/in/axross/","https://www.facebook.com/axrossio","https://www.instagram.com/axrossio/"]}
  </script>,
  <meta
    content="https://tests.kohei.dev/?hl=en-US"
    property="og:url"
  />,
  <meta
    content="profile"
    property="og:type"
  />,
  <meta
    content="Kohei Asai's personal website. My experience, projects and bloggging."
    property="og:description"
  />,
  <meta
    content="en-US"
    property="og:locale"
  />,
  <meta
    content="ja-JP"
    property="og:locale:alternate"
  />,
  <meta
    content="Kohei Asai"
    property="og:site_name"
  />,
  <meta
    content="Kohei Asai"
    property="og:title"
  />,
  <meta
    content="https://tests.kohei.dev/profile.jpg"
    property="og:image"
  />,
]
`;

exports[`<Head> when currentLocale="ja-JP" pushes tags into <head> and they matches with the previous snapshots 1`] = `
Array [
  <meta
    content="width=device-width"
    name="viewport"
  />,
  <meta
    charSet="utf-8"
  />,
  <link
    href="https://tests.kohei.dev/?hl=ja-JP"
    hrefLang="ja-JP"
    rel="alternate"
  />,
  <link
    href="https://tests.kohei.dev/posts/feed.xml?hl=en-US"
    rel="alternate"
    title="Blog post Atom feed (en-US)"
    type="application/atom+xml"
  />,
  <meta
    content="ja-JP"
    property="og:locale:alternate"
  />,
  <title>
    Kohei Asai
  </title>,
  <meta
    content="Kohei Asaiの個人ウェブサイト。経歴や活動状況、ブログなど。"
    name="description"
  />,
  <link
    href="https://tests.kohei.dev/?hl=ja-JP"
    rel="canonical"
  />,
  <link
    href="https://tests.kohei.dev/?hl=en-US"
    hrefLang="en-US"
    rel="alternate"
  />,
  <link
    href="https://tests.kohei.dev/posts/feed.xml?hl=ja-JP"
    rel="alternate"
    title="Blog post Atom feed (ja-JP)"
    type="application/atom+xml"
  />,
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"Person","url":"https://tests.kohei.dev/?hl=ja-JP","name":"Kohei Asai","image":"https://tests.kohei.dev/profile.jpg","jobTitle":"Frontend Engineer","sameAs":["https://github.com/axross","https://linkedin.com/in/axross/","https://www.facebook.com/axrossio","https://www.instagram.com/axrossio/"]}
  </script>,
  <meta
    content="https://tests.kohei.dev/?hl=ja-JP"
    property="og:url"
  />,
  <meta
    content="profile"
    property="og:type"
  />,
  <meta
    content="Kohei Asaiの個人ウェブサイト。経歴や活動状況、ブログなど。"
    property="og:description"
  />,
  <meta
    content="ja-JP"
    property="og:locale"
  />,
  <meta
    content="en-US"
    property="og:locale:alternate"
  />,
  <meta
    content="Kohei Asai"
    property="og:site_name"
  />,
  <meta
    content="Kohei Asai"
    property="og:title"
  />,
  <meta
    content="https://tests.kohei.dev/profile.jpg"
    property="og:image"
  />,
]
`;
